### PR TITLE
Add support for essential workloads

### PR DESCRIPTION
**Pull request by kthomas**
This PR adds support for running "essential" workloads with `nex run` and `nex yeet` using the `--essential` flag.

Essential workloads are automatically redeployed if they exit with a non-zero exit code.

Only `elf` workloads can only be designated as `--essential`.

This PR does not include support for analyzing the `retry_count` or `retried_at` timestamp supplied with requests to redeploy essential workloads, so `elf` processes that exit with a non-zero status very quickly after being started will result in spamming the nex node with redeployments.

### CODE REVIEW

File: internal/control-api/run.go [hunk 1]
```diff
@@ -90,11 +96,6 @@ func (request *DeployRequest) Validate(myKey nkeys.KeyPair) (*jwt.GenericClaims,
 		return nil, fmt.Errorf("workload name claim ('%s') does not match requirements of all lowercase letters", claims.Subject)
 	}
 
-	err = request.DecryptRequestEnvironment(myKey)
```
[the remark this answers is not in the document]
Comment (kthomas) [reply internal/control-api/run.go:93]: Should validation have a side effect of setting properties on the validated instance? We are still ensuring decryption succeeds before starting it, just not hidden within the validate method itself.

File: internal/control-api/run.go [hunk 2]
```diff
@@ -148,6 +149,7 @@ type requestOptions struct {
 	workloadDescription string
 	location            url.URL
 	env                 map[string]string
+	essential           bool
```
[the remark this answers is not in the document]
Comment (kthomas) [reply internal/control-api/run.go:152]: I borrowed "essential" from the AWS naming convention in ECS.

File: internal/control-api/run.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (autodidaddict) [reply internal/control-api/run.go:93]: ok.. as long as we're rejecting deploy requests with unencryptable environments _before_ we attempt to dispatch to an agent... I'm good.

**Review by autodidaddict — APPROVED** [review]
:seal:

File: internal/agent-api/types.go [hunk 3]
```diff
@@ -73,9 +77,22 @@ type DeployRequest struct {
...
+	SenderPublicKey      *string  `json:"-"`
+	TargetNode           *string  `json:"-"`
+	WorkloadJwt          *string  `json:"-"`
+
 	Errors []error `json:"errors,omitempty"`
 }
 
+// Returns true if the run request supports essential flag
+func (request *DeployRequest) SupportsEssential() bool {
+	return strings.EqualFold(*request.WorkloadType, "elf") ||
```
[the remark this answers is not in the document]
Comment (kthomas) [reply internal/agent-api/types.go:92]: I can pull `oci` workload references out of the codebase as part of a separate PR as it shows up elsewhere.

**Pull request merged**